### ISSUE TITLE
Document retrieving body of HttpClientResponseException response

### ISSUE THREAD

**Issue by ivyazmitinov** [micronaut-projects/micronaut-core#2563]
### Steps to Reproduce

1. Create declarative http client with return type of the method as custom POJO class or `retrieve` via low-level client with POJO class as body type argument.
2. Get response with code >= 400 and some body

### Expected Behaviour

`exception.getResponse().getBody()` returns `Optional.empty()`. 
It is actually expected, as erroneous response will most likely differ from successful one, and with `String.class` or `Map.class` instead of POJO class as body type the body is accessible, but there is no documentation  of such behavior, which will be nice to have, if this is how it should behave.

I might update the according docs, if needed.
### Environment Information

- **Micronaut Version:** 1.2.8

**Comment by olevold** [comment]
`exception.getResponse().getBody(ByteBuf.class)` gives you the raw bytes of the error response

**Comment by olevold** [comment]
Or you could use `@Client(errorType = String.class)` to annotate your client, then `exception.getResponse().getBody(String.class)`

**Comment by ivyazmitinov** [comment]
@olevold, yes, it is stated in the description, that the error response is accessable. The issue is not about its' accessability, but about the lack of documentation regarding the ways it can be accessed.

You are more than welcome to update the docs 🙂

**Comment by graemerocher** [comment]
Triage: Type Feature. The current guide says HttpClientResponseException contains the original response, but it does not show the important distinction between decoding it as the successful POJO type, an error type/String/Map, or raw bytes. The existing discussion supplies the workaround, so no new reproducer is needed. Suggested solution: document `@Client(errorType = ...)`, `getResponse().getBody(errorType)`, raw `ByteBuffer` access, and the behavior for blocking, reactive, streaming, and declarative clients.
